src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -83,13 +90,41 @@ export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
     };
 
     public setTcpForwarding = async (deviceId: string): Promise<number> => {
+        const hostPort = await this.portFinder({
+            port: servicePortNumber,
+            stopPort: servicePortNumber + 100,
+        });
+
         this.adb.setDeviceId(deviceId);
-        await this.adb.forwardPort(defaultAdbPortNumber, defaultAdbPortNumber);
-        return defaultAdbPortNumber;
+        await this.adb.forwardPort(hostPort, servicePortNumber);
+        return hostPort;
     };
 
     public removeTcpForwarding = async (deviceId: string): Promise<void> => {
         this.adb.setDeviceId(deviceId);
-        await this.adb.removePortForward(defaultAdbPortNumber);
+        const applicableHostPorts = await this.getHostPortsForwardedToDevicePort(servicePortNumber);
+        for (const applicableHostPort of applicableHostPorts) {
+            await this.adb.removePortForward(applicableHostPort);
+        }
+    };
+
+    // Assumes setDeviceId was already invoked
+    private getHostPortsForwardedToDevicePort = async (devicePort: number): Promise<number[]> => {
```

Comment:
Do we need to worry about the case where multiple instances of the client run simultaneously? There's no theoretical reason that it wouldn't work at the moment, and we do nothing to prevent it. Cleaning up all ports redirected to the device port would impact this multiple instance scenario. As soon as any of the instances exits, all of the instances lose their port forwarding. That feels a little bit aggressive--I think I'd prefer to see us remove just the port that we added within this instance.

Reply:
That's a fair point, I hadn't considered running multiple copies of the app. Will update to have it take a hostPort param instead